apm-agent-core/src/main/java/co/elastic/apm/agent/impl/EmptyElasticContext.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.impl;
+
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.ElasticContext;
+import co.elastic.apm.agent.tracer.Scope;
+
+import javax.annotation.Nullable;
+
+class EmptyElasticContext extends ElasticContext<EmptyElasticContext> {
```

Comment:
This is unfortunately not really possible via inheritance because the implementation variant of `ElasticContext` is now a class, so `EmptyElasticContext` can't inherit from both `NoopElasticContext` and `ElasticContext`. We could do it via composition instead, but this is not worth it here imo. In addition in a follow up PR this class wil lbecome even smaller: In the follow up PR I've moved the activation method implementations up into `ElasticContext`. The main reason is that `EmptyElasticContext.activate()` is a bit flawed with the current implementation: If a span/transaction/baggage is currently active and you invoke `EmptyElasticContext.activate()`, it doesn't "clear" the currently active span/transaction/baggage. This is nothing we do in production code, but it is useful in tests and otherwise also counterintuitive.